### PR TITLE
Produce activitysim skims at runtime

### LINKED ISSUE #3209: Run ActivitySim with Beam using the new Merging feature
Merging feature: #3192

### PR DESCRIPTION
**Pull request by dimaopen**
Example config 
```
beam.router.skim = {
  writeSkimsInterval = 1
}

beam.exchange{
  output {
    activitySimSkimsEnabled = true
    # geo level different than TAZ (in beam taz-centers format)
#    geo.filePath = ${beam.inputDirectory}"/block_group-centers.csv.gz"
  }
}
```

Resolves #3209

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3238)

### CODE REVIEW

**Review by alex-vv — APPROVED** [review]

**Comment by dimaopen** [comment]
Test!

File: src/main/scala/beam/router/skim/ActivitySimSkimmer.scala [hunk 1]
```diff
@@ -240,39 +319,49 @@ object ActivitySimSkimmer extends LazyLogging {
...
 
-    def toCsvString: String = {
-      s"$timePeriodString,$pathType,$originId," +
-      s"$destinationId,$weightedGeneralizedTime,$weightedTotalInVehicleTime," +
-      s"$weightedGeneralizedCost,$weightedDistance,$weightedWalkAccess," +
-      s"$weightedWalkAuxiliary,$weightedWalkEgress,$weightedDriveTimeInMinutes," +
-      s"$weightedDriveDistanceInMeters,$weightedLightRailInVehicleTimeInMinutes,$weightedFerryInVehicleTimeInMinutes," +
-      s"$weightedTransitBoardingsCount,$weightedCost,$debugText\n"
-    }
+    def toCsvString: String = productIterator.mkString("", ",", "\n")
```
Comment (JustinPihony) [inline src/main/scala/beam/router/skim/ActivitySimSkimmer.scala:337]: If moving to utilizing the `productIterator` then we should add a test to make sure any future changes are deliberate.

File: src/main/scala/beam/utils/csv/CsvWriter.scala [hunk 2]
```diff
@@ -46,9 +46,10 @@ class CsvWriter(
     Try(writer.close())
   }
 
-  def writeAllAndClose(rows: Iterable[Seq[Any]]): Unit = {
-    rows.foreach(writeRow)
+  def writeAllAndClose(rows: Iterable[Seq[Any]]): Try[Unit] = {
+    val result = Try(rows.foreach(writeRow))
```
[the remark this answers is not in the document]
Comment (dimaopen) [reply src/main/scala/beam/utils/csv/CsvWriter.scala:50]: This method is intended to just save all the data we have without the possibility to recover. But it also will not terminate the run if something goes wrong here.

**Comment by dimaopen** [comment]
Test!

**Comment by dimaopen** [comment]
Test!

**Pull request merged**